From 49cb4039be99d2d49f5c97edd22fc47976c34651 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Niels=20M=C3=B6ller?= <nisse@lysator.liu.se>
Date: Sun, 10 Jan 2021 20:39:49 +0100
Subject: [PATCH] ppc: Fix use of __GLIBC_PREREQ in fat-ppc.c.

* fat-ppc.c: Don't use __GLIBC_PREREQ in the same preprocessor
conditional as defined(__GLIBC_PREREQ), but move to a nested #if
conditional. Fixes compile error on OpenBSD/powerpc64, reported by
Jasper Lievisse Adriaanse.
---
 ChangeLog |  7 +++++++
 fat-ppc.c | 11 ++++++-----
 2 files changed, 13 insertions(+), 5 deletions(-)

diff --git a/ChangeLog b/ChangeLog
index 578f2591..a5506897 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,3 +1,10 @@
+2021-01-10  Niels Möller  <nisse@lysator.liu.se>
+
+	* fat-ppc.c: Don't use __GLIBC_PREREQ in the same preprocessor
+	conditional as defined(__GLIBC_PREREQ), but move to a nested #if
+	conditional. Fixes compile error on OpenBSD/powerpc64, reported by
+	Jasper Lievisse Adriaanse.
+
 2021-01-04  Niels Möller  <nisse@lysator.liu.se>
 
 	* Released Nettle-3.7.
diff --git a/fat-ppc.c b/fat-ppc.c
index c7f0d11a..3adbb88c 100644
--- a/fat-ppc.c
+++ b/fat-ppc.c
@@ -42,11 +42,12 @@
 
 #if defined(_AIX)
 # include <sys/systemcfg.h>
-#elif defined(__linux__) && defined(__GLIBC__) && \
-  defined(__GLIBC_PREREQ) && __GLIBC_PREREQ(2, 16)
-# define USE_GETAUXVAL 1
-# include <asm/cputable.h>
-# include <sys/auxv.h>
+#elif defined(__linux__) && defined(__GLIBC__) && defined(__GLIBC_PREREQ)
+# if __GLIBC_PREREQ(2, 16)
+#  define USE_GETAUXVAL 1
+#  include <asm/cputable.h>
+#  include <sys/auxv.h>
+# endif
 #elif defined(__FreeBSD__)
 # include <machine/cpu.h>
 # ifdef PPC_FEATURE2_HAS_VEC_CRYPTO
-- 
GitLab

